pkg/cmd/roachtest/tests/cdc.go

```diff
@@ -1913,6 +1921,56 @@ func registerCDC(r registry.Registry) {
 			runCDCKafkaAuth(ctx, t, c)
 		},
 	})
+	r.Add(registry.TestSpec{
+		Name:             "cdc/kafka-auth-msk",
+		Owner:            `cdc`,
+		Cluster:          r.MakeClusterSpec(1),
+		Leases:           registry.MetamorphicLeases,
+		CompatibleClouds: registry.OnlyAWS,
+		Suites:           registry.Suites(registry.Nightly),
+		RequiresLicense:  true,
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			c.Start(ctx, t.L(), option.DefaultStartOpts(), install.MakeClusterSettings(), c.All())
+			mkm := NewMskManager(ctx, t, msktypes.ClusterTypeServerless)
+			mkm.MakeCluster(ctx)
+			defer mkm.Teardown()
+
+			t.Status("waiting for msk cluster to be active")
+			waitCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+			brokers := mkm.WaitForClusterActiveAndDNSUpdated(waitCtx, c)
+			cancel()
+			t.Status("cluster is active")
+			mkm.CreateTopic(ctx, "auth_test_table", c)
+
+			db := c.Conn(ctx, t.L(), 1)
+			defer stopFeeds(db)
+
+			tdb := sqlutils.MakeSQLRunner(db)
+			tdb.Exec(t, `CREATE TABLE auth_test_table (a INT PRIMARY KEY)`)
+
+			testCerts, err := makeTestCerts("0.0.0.0", "aws") // ?
+			require.NoError(t, err)
+			feeds := map[string]string{
+				"scram": fmt.Sprintf("kafka://%s?tls_enabled=true&ca_cert=%s&sasl_enabled=true&sasl_user=scram512&sasl_password=scram512-secret&sasl_mechanism=SCRAM-SHA-512", brokers.scram, testCerts.CACertBase64()),
+				"iam":   fmt.Sprintf("kafka://%s?tls_enabled=true&sasl_enabled=true&sasl_mechanism=AWS_MSK_IAM&sasl_aws_region=us-east-2&sasl_aws_iam_role_arn=arn:aws:iam::541263489771:role/miles-msk-testing&sasl_aws_iam_session_name=miles", brokers.iam),
```

Comment:
TODO: make a new iam role with the same permissions but a different name and use that